homeassistant/components/fritz/common.py

```diff
@@ -512,6 +524,13 @@ async def service_fritzbox(
                 await self.async_trigger_cleanup(config_entry)
                 return
 
+            if service_call.service == SERVICE_SET_GUEST_WIFI_PW:
+                await self.async_trigger_set_guest_password(
+                    service_call.data.get("password"),
+                    service_call.data.get("length"),
```

Comment:
use `DEFAULT_PASSWORD_LENGTH` from `fritzconnection.lib.fritzwlan` as default

```suggestion
                    service_call.data.get("length", DEFAULT_PASSWORD_LENGTH),
```
the `length` parameter in `fritz_guest_wifi.set_password` needs to be an `int`

Reply:
But if `None` is passed to the `set_password` method it already has the `DEFAULT_PASSWORD_LENGHT` as a default parameter. So this should not be necessary?!

Reply:
https://github.com/kbr/fritzconnection/blob/master/fritzconnection/lib/fritzwlan.py#L252

Reply:
You explicit set the `length` to the result of `service_call.data.get("length")` ... so if it returns `None` than `length` will be set to None

Reply:
Right...silly me :-/